### DISCUSSION TITLE
Why is @esbuild-plugins/node-globals-polyfill not being used?

### DISCUSSION THREAD

**Discussion by aiji42** [remix-run/remix#2566] [Q&A]
This discussion is derived from this one https://github.com/remix-run/remix/discussions/2437#discussioncomment-2471938

### Question
**Is there any reason why Remix is not using @esbuild-plugins/node-globals-polyfill in the compiler?**

I would like to know if anyone knows, the compiler uses `@esbuild-plugins/node-modules-polyfill` but not `@esbuild-plugins/node-globals-polyfill`.
https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler.ts#L7

### Why I want to know this.

Many styling libraries use a global Buffer in their modules for the server.
As far as I know, styling libraries such as emotion, chakra-ui, mantine, etc. depend on global buffer and do not work with cloudflare. Using `@esbuild-plugins/node-globals-polyfill` with the compiler is [guaranteed to solve this problem](https://github.com/aiji42/remix-esbuild-override#examples).

**Comment by jdnichollsc** [comment]
Discussion related #3203
